go-controller/pkg/clustermanager/endpointslicemirror/endpointslice_mirror_controller.go

```diff
@@ -88,7 +88,12 @@ func (c *Controller) getDefaultEndpointSliceKey(endpointSlice *v1.EndpointSlice)
 }
 
 func (c *Controller) enqueueEndpointSlice(obj interface{}) {
-	if key := c.getDefaultEndpointSliceKey(obj.(*v1.EndpointSlice)); key != "" {
+	eps, ok := obj.(*v1.EndpointSlice)
```

Comment:
shouldn't we be handling tombstone state?

like:

```
endpointSlice, ok := obj.(*discovery.EndpointSlice)
	if !ok {
		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
			return
		}
		endpointSlice, ok = tombstone.Obj.(*discovery.EndpointSlice)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a EndpointSlice: %#v", obj))
			return
		}
	}

	if endpointSlice != nil {
		c.queueServiceForEndpointSlice(endpointSlice)
	}
```

```
// DeletedFinalStateUnknown is placed into a DeltaFIFO in the case where an object
// was deleted but the watch deletion event was missed while disconnected from
// apiserver. In this case we don't know the final "resting" state of the object, so
// there's a chance the included `Obj` is stale.
type DeletedFinalStateUnknown struct {
	Key string
	Obj interface{}
}
````

Dont we still want to try to process this object if we missed it and it may need deletion?